recipes/azure-core-cpp/meta.yaml

```diff
@@ -0,0 +1,65 @@
+{% set version = "1.10.0" %}
+{% set sha256 = "e6ae26ad30376bef6272024564764ba7bf5756063dc94cc4b74180cd004ab75e" %}
+
+package:
+  name: azure-core-cpp
+  version: {{ version }}
+
+source:
+  url: https://github.com/Azure/azure-sdk-for-cpp/archive/refs/tags/azure-core_{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    # https://github.com/Azure/azure-sdk-for-cpp/issues/4784
+    # https://github.com/Azure/azure-sdk-for-cpp/pull/4785
+    - 0001-remove-wil-from-exported-config.patch
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("azure-core-cpp", min_pin="x", max_pin="x") }}
+
+# https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/vcpkg.json
+requirements:
+  build:
+    - {{ compiler("cxx") }}
+    - cmake
+    - ninja
+  host:
+    - libcurl
+    - openssl  # [not win]
+    - wil  # [win]
+
+test:
+  commands:
+    # shared libs
+    - test -f ${PREFIX}/lib/libazure-core.so            # [linux]
+    - test -f ${PREFIX}/lib/libazure-core.dylib         # [osx]
+    - if not exist %LIBRARY_BIN%\azure-core.dll exit 1  # [win]
+    - if not exist %LIBRARY_LIB%\azure-core.lib exit 1  # [win]
+
+    # headers
+    - test -f ${PREFIX}/include/azure/core.hpp    # [unix]
+    - if not exist %LIBRARY_INC%\azure\core.hpp exit 1  # [win]
+
+    # CMake metadata
+    - test -f ${PREFIX}/share/azure-core-cpp/azure-core-cppTargets.cmake                     # [unix]
+    - if not exist %LIBRARY_PREFIX%\share\azure-core-cpp\azure-core-cppTargets.cmake exit 1  # [win]
```

Comment:
To know if a library CMake configuration can be used, and to document its use, those tests could be completed with the build of a simple CMake project using this library.

Here is [a simple example of such tests](https://github.com/conda-forge/staged-recipes/pull/23539/files#diff-0287fef753ebf619329766f0ff34c4fe490a07b4d759d9b6e7063d6118661610R36-R42).

This suggestion also applies to the other libraries' recipe.

Reply:
Thank you!

In this case, I think you can use a simpler CMake setup such as, for `azure-identity-cpp`:

```cmake
cmake_minimum_required(VERSION 3.0)
project(azure-identity-cpp-dependent-project LANGUAGES C CXX)

find_package(azure-identity-cpp REQUIRED)
```

Reply:
Ah, I see. You mainly want to test that `find_package()` can find the installed CMake configuration files. I'll leave my more comprehensive test of azure-core-cpp (ie that includes `target_link_libraries()`), but then use the simpler `CMakeLists.txt` for the other libraries.